web: fix navbar button styling

landism opened the pull request:
Fixes navbar issues from this [PR's comment](https://github.com/tilt-dev/tilt/pull/4898#pullrequestreview-742973668):
1. title display on hover overlaps and can lead the user to click the wrong button
2. (very subtle) drop shadow on buttons that shouldn't be there
3. ripple displays on click and shouldn't be there
4. icon and button seems to darken/flicker on click, though this could be related to the disabled/loading state
5. the click area is a little strange for the button groups. the click area is large for the command button itself, but small for the dropdown button. i think we could solve this by making the size of the command button / button group smaller and centering the button title display, but i didn't play around with this solution!

lizzthabet commented on web/src/ApiButton.tsx:
```diff
@@ -281,7 +281,10 @@ export function ApiButton(props: React.PropsWithChildren<ApiButtonProps>) {
+  const onClick = async (e: React.MouseEvent<HTMLElement>) => {
+    if ((e.target as Element).classList.contains("ignoreClick")) {
+      return true
+    }
```
it seems a little fishy to me to cancel a button click if it happens on a certain area of a button. (to me, that's an indication that maybe html structure and css styling could be improved, and i'd be a little bit concerned with ux or a11y ux implications.) is there a reason you went with this approach vs a css layout adjustment?

lizzthabet commented:
ps. other changes/fixes look good so far! i just want to make sure i understand the overlapping text fix before suggesting an alternative approach (in case you had already investigated alternatives).

landism commented:
OK, think it's good now.

I'm still not thrilled with the naming...

I figured I'd solve it using MUI's <Tooltip> as a model, which would probably lead to, like:
```
<MenuButtonLabel label="foo">
  <MenuButton .../>
</MenuButtonLabel>
```

but then we need a way to set the opacity on the actual label, which we'd do by making a styled component for the label and referencing it from `MenuButtonLabel`'s css. But the natural name for that label is `MenuButtonLabel`, so one of the two has to change!

So, I changed the outer one to `MenuButtonLabeled`.

(this is also all kind of awkward because MUI's `Tooltip` naming feels weird as well, so being consistent with it is necessarily weird! I looked at its code and it has a bunch of comments like "tooltip (label wrapper) element" to deal with this ambiguity (i.e., does "tooltip" refer to the wrapper component or the actual little tooltip that pops up?)

lizzthabet reviewed (approved):
> So, I changed the outer one to `MenuButtonLabeled`.
> 
> (this is also all kind of awkward because MUI's `Tooltip` naming feels weird as well, so being consistent with it is necessarily weird! I looked at its code and it has a bunch of comments like "tooltip (label wrapper) element" to deal with this ambiguity (i.e., does "tooltip" refer to the wrapper component or the actual little tooltip that pops up?)

maybe `MenuButtonWithLabel` is less awkward? i don't have any strong opinions re:mui structural patterns, but i think this is a better solution than the "ignore click here" approach! this is definitely an improvement over the overlapping labels on hover. 🎉 

things i think we could address later:

* i'd like to improve the a11y markup on the nav buttons. i think this would be something like:

```html
<!-- button group -->
  <div role="group" aria-label={label}>
    <button aria-label={`Trigger ${label}`}></button>
    <button aria-label={`Open ${label} options`}></button>
  </div>
<!-- regular button -->
  <button aria-label={label}></button>
```
(also would be happy to pair on this or put up a pr later!)

* long button names could be trimmed. this is pretty minor and perhaps only worth address if we hear it's a user pain point! 
![Screen Shot 2021-09-08 at 11 23 19 AM](https://user-images.githubusercontent.com/23283986/132543504-d835fb24-dc90-4f9d-9173-2c48528f2c0a.png)

lizzthabet commented:
oh! i just noticed that the sizing for button groups in table view is wonky in this branch only:
![Screen Shot 2021-09-08 at 12 33 39 PM](https://user-images.githubusercontent.com/23283986/132549088-6d182b5a-eb1a-42be-b953-bb731de7a4bd.png)

landism commented on web/src/CustomNav.tsx:
```diff
@@ -1,18 +1,30 @@
+    box-shadow: none;
+    justify-content: center;
+
+    &:hover {
+      box-shadow: none;
+    }
+  }
```
The `&:active` seemed to cover it and take priority over `.MuiButton-contained:active` - did you hit a case where the latter is needed?

lizzthabet replied:
oh sweet! that should be fine then. i think i started with the `.Mui`-specific fix, then added the more general one later when i was testing.

landism commented on web/src/GlobalNav.tsx:
```diff
@@ -64,14 +65,16 @@ export const MenuButtonMixin = `
 `
+export const MenuButtonLabeledRoot = styled.div`
+  position: relative; // Anchor MenuButtonLabel, which shouldn't affect this element's width
+  &:hover ${MenuButtonLabel}, button[data-open="true"] + ${MenuButtonLabel} {
```
done!

landism commented:
> oh! i just noticed that the sizing for button groups in table view is wonky in this branch only
fixed

> i'd like to improve the a11y markup on the nav buttons
done

I also noticed the vertical alignment on the tilt version button was off a bit so I fixed that
